index.js

```diff
@@ -123,26 +126,28 @@ ExpressOAuthServer.prototype.token = function(options) {
 
 var handleResponse = function(req, res, response) {
   res.set(response.headers);
-
-  res.send(response.body);
-  res.status(response.status);
+  res.status(response.status).send(response.body);
 };
 
 /**
  * Handle error.
  */
 
-var handleError = function(e, req, res, response) {
-  if (response) {
-    res.set(response.headers);
-  }
+var handleError = function(e, req, res, response, next) {
 
-  if (e instanceof UnauthorizedRequestError) {
-    return res.status(e.code);
-  }
+  if (this.useErrorHandler === true) {
```

Comment:
Down here, `this === global`.

We need to either pass `useErrorHandler` as an argument or invoke `handleError` (and to be consistent `handleResponse`) using `handleError.call(this, ...)`.